alibi/explainers/cfrl_base.py

```diff
@@ -318,68 +331,72 @@ def sample(self) -> Dict[str, Optional[np.ndarray]]:
 
 
 class CounterfactualRLBase(Explainer, FitMixin):
-    """ Counterfactual Reinforcement Learning Base. """
-
-    PYTORCH = "pytorch"
-    TENSORFLOW = "tensorflow"
+    """
+     Counterfactual Reinforcement Learning Base.
+     TODO: check if all tensors are upperca
+    """
 
     def __init__(self,
-                 predict_func: Callable,
-                 ae,
+                 predictor: Callable,
+                 ae: Union['TensorflowAE', 'PytorchAE'],
```

Comment:
Just to confirm, we're going for an API that takes `encoder`, `decoder` as native `tensorflow` or `pytorch` types ?

Reply:
Yes. I will modify it to encoder and decoder.

Reply:
Would this require defining an encoder/decoder wrapper for each backend?

Reply:
The type hints can be native types, i.e. `tf.keras.Layer`, `nn.Module`, they just need to be imported within the `TYPE_CHECKING` guard.